### PR TITLE
Reduced time complexity for reversing a string

### PR DESCRIPTION
**Pull request by pritam-dash**
Added a third way of reversing a string, which reduces the time complexity of the code.

### **Describe your change:**

- [x] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [ ] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/main/java/com/thealgorithms/strings/ReverseString.java [hunk 1]
```diff
@@ -39,4 +40,28 @@ public static String reverse2(String str) {
...
+     */
+    public static String reverse3(String str) {
+        if (str == null || str.isEmpty()) {
+            return str;
+        }
+
+        int len=str.length();
+        String s1="",s2="";
+        for(int i=0, j=len-1; i<len/2; i++, j--)
+        {
```
Comment (Teuton-Roy) [inline src/main/java/com/thealgorithms/strings/ReverseString.java:58]:
```suggestion
        for(int i=0, j=len-1; i<len/2; i++, j--) {
```

File: src/main/java/com/thealgorithms/strings/ReverseString.java [hunk 2]
```diff
@@ -39,4 +40,28 @@ public static String reverse2(String str) {
...
+            s1=str.charAt(i)+s1;
+            s2+=str.charAt(j);
+        }
+        if(len%2==0)
+        return new String(s2+s1);
+        else
+        return new String(s2+str.charAt(len/2)+s1);
```
Comment (Teuton-Roy) [inline src/main/java/com/thealgorithms/strings/ReverseString.java:65]:
```suggestion
        if(len%2 == 0)
            return new String(s2 + s1);
        else
            return new String(s2 + str.charAt(len/2) + s1);
```

**Review by Teuton-Roy — APPROVED** [review]
Just need little bit of formating

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please add some automated tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by pritam-dash** [comment]
Can you kindly elaborate on the upgrades that you are looking for?
It would help me to further integrate the desired tests with curated results.

**Comment by siriak** [comment]
JUnit tests https://www.tutorialspoint.com/junit/junit_test_framework.htm

**Pull request closed without merging**